src/Node.h

```diff
@@ -103,7 +103,11 @@ class WIN32_EXPORT Node {
             return nullptr;
         }
 
-        ::connect(fd, result->ai_addr, result->ai_addrlen);
+        if (::connect(fd, result->ai_addr, result->ai_addrlen) == -1) {
```

Comment:
-1 should (pedantically) be SOCKET_ERROR (even though it is really just a macro for -1) on Windows. Or != 0.